tasks/main.yml

```diff
@@ -36,29 +36,14 @@
 - name: install docker-rotate cron script
   template: src=docker-rotate.j2 dest=/etc/cron.daily/docker-rotate mode=755
 
-- name: push logrotate config file for docker container json logs
-  template: src=logrotate.conf.j2 dest=/etc/logrotate.d/docker
-  when: installed_docker_version | version_compare('1.8', '<')
-
+# docker log rotation is now managed by Docker (since Docker 1.8). We no longer deal with
```

Comment:
We haven't migrated to docker 1.8+ everywhere, but...

The way to manage docker log rotation now, for post 1.8, is to just have your project pass in log config options when it sets docker_daemon_flags. That's a lot cleaner than having special code in "docker-base" or "docker" to parse log config variables and convert them into daemon flags, for a couple of reasons:
 - there's just less code
 - log rotation config is now as documented in the Docker manual, not using a separate convention we invented here
 - there's less coupling - the code we had before made assumptions about the log driver used, and might have caused trouble if we moved to another log driver.

This does leave us in a weird place for pre-1.8 docker. Our options are:
 - have this role support log rotation, but only for pre-1.8 docker - the functionality wouldn't work for post-1.8.
 - create a separate logrotate role for docker <1.8.
 - plan for <1.8 systems to use old versions of these roles.

I chose the third option. The first is weird, and we run the risk that someone upgrading docker will miss the change. The second is extra work that I think is unnecessary. The third is fine - our internal systems that are still on docker 1.5 are already locked to an older versions of these roles.

Documentation is a bit more complex. We could say that the role doesn't support Docker <1.8, but that's technically not true - all the currently supported functionality of the role will work fine with old Docker. I think I need to add a comment to README and CHANGES that addresses the point in detail.